### PR TITLE
Planning:frame for open space planner

### PR DESCRIPTION
**Pull request by JasonZhou404**

### CODE REVIEW

File: modules/planning/common/frame_open_space.cc [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+using apollo::common::VehicleStateProvider;
+using apollo::common::adapter::AdapterManager;
+using apollo::common::math::Box2d;
+using apollo::common::math::Vec2d;
+using apollo::common::monitor::MonitorLogBuffer;
+using apollo::prediction::PredictionObstacles;
+
+constexpr double kMathEpsilon = 1e-8;
+
+Frame_open_space_History::Frame_open_space_History()
```
Comment (Capri2014) [inline modules/planning/common/frame_open_space.cc]: Use "FrameOpenSpaceHistory" instead

File: modules/planning/common/frame_open_space.cc [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+  return nullptr;
+}
+
+uint32_t Frame_open_space::SequenceNum() const { return sequence_num_; }
+
+std::string Frame_open_space::DebugString() const {
+  return "Frame_open_space: " + std::to_string(sequence_num_);
+}
+
+void Frame_open_space::RecordInputDebug(planning_internal::Debug *debug) {
```
Comment (Capri2014) [inline modules/planning/common/frame_open_space.cc]: Remove routing related information

File: modules/planning/common/frame_open_space.cc [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+  auto *adc_position = planning_data->mutable_adc_position();
+  const auto &localization =
+      AdapterManager::GetLocalization()->GetLatestObserved();
+  adc_position->CopyFrom(localization);
+
+  const auto &chassis = AdapterManager::GetChassis()->GetLatestObserved();
+  auto debug_chassis = planning_data->mutable_chassis();
+  debug_chassis->CopyFrom(chassis);
+
+  if (!FLAGS_use_navigation_mode) {
```
Comment (Capri2014) [inline modules/planning/common/frame_open_space.cc]: And navigation mode related information

File: modules/planning/common/frame_open_space.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+class Frame_open_space_History : public IndexedQueue<uint32_t, Frame_open_space> {
+ private:
+  DECLARE_SINGLETON(Frame_open_space_History);
+};
+
+}  // namespace planning
+}  // namespace apollo
+
+#endif  // MODULES_PLANNING_COMMON_FRAME_OPEN_SPACE_H_
```
Comment (jinghaomiao) [inline modules/planning/common/frame_open_space.h]: We are enforcing the unit test. Please submit the corresponding unit test.

**Comment by JasonZhou404** [comment]
Unit test added, a few test will be carried out easier with more implementation of the open space planner @lianglia-apollo @jinghaomiao

**Pull request merged**